bin/add-entry-to-tsconfig.js

```diff
@@ -0,0 +1,17 @@
+const fs = require('fs')
+
+const { parse, stringify } = require('comment-json')
+
+let component_name = process.argv[2]
+
+if (component_name === undefined)
+  throw new Error('please provide component name')
+
+let tsconfig = parse(fs.readFileSync('./tsconfig.json', 'utf8'))
+
+tsconfig.include.push(
+  `src/components/${component_name}/${component_name}.vue`
+)
+fs.writeFileSync('./tsconfig.json', stringify(tsconfig, null, 3), { flag: 'w' })
```

Comment:
We use 2 spaces for indentation. Using 3 here causes the entire file to output with changes on every line.

```suggestion
fs.writeFileSync('./tsconfig.json', stringify(tsconfig, null, 2), { flag: 'w' })
```

I also noticed that doing it this way (with `comment-json`) creates other changes in the file that would need to be fixed with the linter:

```diff
diff --git a/tsconfig.json b/tsconfig.json
index b8e8fe59f..5e1c95475 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -6,7 +6,11 @@
     "jsx": "preserve",
     "target": "ES2018",
     "module": "ESNext",
-    "lib": ["DOM", "ESNext", "ESNext.AsyncIterable"],
+    "lib": [
+      "DOM",
+      "ESNext",
+      "ESNext.AsyncIterable"
+    ],
     "noEmit": true,
     "isolatedModules": true,
     "skipLibCheck": true,
@@ -22,11 +26,23 @@
     /* Module Resolution Options */
     "moduleResolution": "Node",
     "resolveJsonModule": true,
-    "types": ["@types/jest", "@nuxt/types", "@nuxtjs/i18n", "@pinia/nuxt"],
-    "typeRoots": ["./typings", "./node_modules/@types"],
+    "types": [
+      "@types/jest",
+      "@nuxt/types",
+      "@nuxtjs/i18n",
+      "@pinia/nuxt"
+    ],
+    "typeRoots": [
+      "./typings",
+      "./node_modules/@types"
+    ],
     "paths": {
-      "~/*": ["./src/*"],
-      "~~/*": ["./*"]
+      "~/*": [
+        "./src/*"
+      ],
+      "~~/*": [
+        "./*"
+      ]
     },
     "baseUrl": "."
   },
@@ -81,7 +97,6 @@
     "src/components/VHeader/VPageLinks.vue",
     "src/components/VHeader/VSearchBar/**.vue",
     "src/components/VImageDetails/VRelatedImages.vue",
-
     "src/pages/about.vue",
     "src/pages/feedback.vue",
     "src/pages/search-help.vue",
@@ -93,10 +108,13 @@
      * with other tools we rely on.
      */
     "src/constants/**/*",
-    "src/server-middleware/healthcheck.js"
+    "src/server-middleware/healthcheck.js",
     /** End of non TypeScript files section */
+    "src/components/VSarasComponent/VSarasComponent.vue"
+  ],
+  "exclude": [
+    "node_modules"
   ],
-  "exclude": ["node_modules"],
   "vueCompilerOptions": {
     "experimentalCompatMode": 2,
     "experimentalTemplateCompilerOptions": {
@@ -106,4 +124,4 @@
     },
     "target": 2.7
   }
-}
+}
\ No newline at end of file
```

The linter fixes and catches all of these but it causes the initial attempt at committing to fail due to the new changes:

```diff
diff --git a/tsconfig.json b/tsconfig.json
index 5e1c95475..459484e0e 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -6,11 +6,7 @@
     "jsx": "preserve",
     "target": "ES2018",
     "module": "ESNext",
-    "lib": [
-      "DOM",
-      "ESNext",
-      "ESNext.AsyncIterable"
-    ],
+    "lib": ["DOM", "ESNext", "ESNext.AsyncIterable"],
     "noEmit": true,
     "isolatedModules": true,
     "skipLibCheck": true,
@@ -26,23 +22,11 @@
     /* Module Resolution Options */
     "moduleResolution": "Node",
     "resolveJsonModule": true,
-    "types": [
-      "@types/jest",
-      "@nuxt/types",
-      "@nuxtjs/i18n",
-      "@pinia/nuxt"
-    ],
-    "typeRoots": [
-      "./typings",
-      "./node_modules/@types"
-    ],
+    "types": ["@types/jest", "@nuxt/types", "@nuxtjs/i18n", "@pinia/nuxt"],
+    "typeRoots": ["./typings", "./node_modules/@types"],
     "paths": {
-      "~/*": [
-        "./src/*"
-      ],
-      "~~/*": [
-        "./*"
-      ]
+      "~/*": ["./src/*"],
+      "~~/*": ["./*"]
     },
     "baseUrl": "."
   },
@@ -112,9 +96,7 @@
     /** End of non TypeScript files section */
     "src/components/VSarasComponent/VSarasComponent.vue"
   ],
-  "exclude": [
-    "node_modules"
-  ],
+  "exclude": ["node_modules"],
   "vueCompilerOptions": {
     "experimentalCompatMode": 2,
     "experimentalTemplateCompilerOptions": {
@@ -124,4 +106,4 @@
     },
     "target": 2.7
   }
-}
\ No newline at end of file
+}
```

I'm not sure what the best way to avoid this is. The only thing I can think of is to directly modify the file as a string and then write it back, without parsing and strigifying the JSON back out. If we did this, I'd probably recommend just adding the new component to the front of the list to make it easier to find the consistent insertion point.

Reply:
@sarayourfriend i`ll try that aproach.  regarding the create-component.sh, i`ll add some input validation in this PR

Reply:
What if we add `eslint --fix tsconfig.json` to the end of the `create:component-scaffolding` command?